sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -387,3 +387,67 @@ case class MinOf(left: Expression, right: Expression) extends BinaryArithmetic {
   override def symbol: String = "min"
   override def prettyName: String = symbol
 }
+
+case class Pmod(left: Expression, right: Expression) extends BinaryArithmetic {
+
+  override def symbol: String = "pmod"
+
+  protected def checkTypesInternal(t: DataType) =
+    TypeUtils.checkForNumericExpr(t, "pmod")
+
+  override lazy val resolved =
+    childrenResolved && checkInputDataTypes().isSuccess && !DecimalType.isFixed(dataType)
+
+  protected override def nullSafeEval(left: Any, right: Any) =
+    dataType match {
```

Comment:
Actually there's not too much performance difference between those coding styles. Please see: https://gist.github.com/zhichao-li/71cd2b82f6a7bad36a3c . `( output: A: 1141  B: 697(current) C:856 D:834) ` and I guess function object with lazy modifier might be a little bit slower since it introduce sort of double check singleton